Application/EdFi.Ods.Api/Middleware/RequestResponseDetailsLoggerMiddleware.cs

```diff
@@ -55,44 +134,66 @@ private static string TryGetStatusCodeDescription(int statusCode)
+
+            RequestResponseContentLogger.Error(restError.Message, exception);
+        }
+
+        private void LogRequestResponseDetailsInfo(HttpContext context)
         {
             // Check the Appender exists to avoid duplicated log messages on both controller(root) and detailed loggers
             if (!RequestResponseDetailsFileAppenderExists())
                 return;
 
-            AddLoggerProperties(context.Request.Method, context.Response.StatusCode, context);
-
             RequestResponseDetailsLogger.Info(TryGetStatusCodeDescription(context.Response.StatusCode));
         }
 
-        private void LogRequestResponseDetailsOnException(HttpContext context, Exception exception)
+        private void LogRequestResponseDetailsOnException(Exception exception)
         {
             // Check the Appender exists to avoid duplicated log messages on both controller(root) and detailed loggers
             if (!RequestResponseDetailsFileAppenderExists())
                 return;
 
             var restError = _restErrorProvider.GetRestErrorFromException(exception);
 
-            AddLoggerProperties(context.Request.Method, restError.Code, context);
-
-            RequestResponseDetailsLogger.Error(restError.Message);
+            RequestResponseDetailsLogger.Error(restError.Message, exception);
         }
 
         protected ILog RequestResponseDetailsLogger
         {
             get => _requestResponseDetailsLogger ??= LogManager.GetLogger("RequestResponseDetailsLogger");
         }
 
-        private void AddLoggerProperties(string requestMethod, int responseCode, HttpContext context)
+        protected ILog RequestResponseContentLogger
+        {
+            get => _requestResponseContentLogger ??= LogManager.GetLogger("RequestResponseContentLogger");
+        }
+
+        private void AddLoggerDetailProperties(string requestMethod, int responseCode, HttpContext context)
         {
             LogicalThreadContext.Properties["RequestUrl"] = context.Request.ResourceUri(_reverseProxySettings);
             LogicalThreadContext.Properties["RequestMethod"] = requestMethod;
             LogicalThreadContext.Properties["ResponseCode"] = responseCode;
         }
 
+        private void AddLoggerContentProperties(string requestBodyContent, string responseBodyContent, HttpContext context)
+        {
+            LogicalThreadContext.Properties["RequestUrlWithQueryString"] = $"{context.Request.ResourceUri(_reverseProxySettings)}{context.Request.QueryString}";
+            LogicalThreadContext.Properties["RequestBody"] = requestBodyContent;
+            LogicalThreadContext.Properties["ResponseBody"] = responseBodyContent;
+        }
+
         private bool RequestResponseDetailsFileAppenderExists() => 
             RequestResponseDetailsLogger.Logger.Repository.GetAppenders().Any(x => x.Name == "RequestResponseDetailsFileAppender");
 
+        private bool RequestResponseContentDatabaseAppenderExists() =>
```

Comment:
@mjaksn This condition is not sufficient. Content logging should be enabled only if database appender is configured.

Reply:
@vimayya please take a look at the new commit on this branch and let me know if it addresses this issue